source/fundamentals/authentication.txt

```diff
@@ -142,6 +143,180 @@ string as follow:
+the following steps:
+
+1. Specify the authentication mechanism and authentication source
+#. Set the appropriate environment variables
+
+You can specify the authentication mechanism and authentication source 
+by using a ``MongoCredential`` or as part of the connection string.
+
+To specify the authentication mechanism and authentication source by using a 
+``MongoCredential``, create a ``MongoCredential`` instance and add it to your 
+``MongoClient`` as shown in the following example:
+
+.. code-block:: csharp
+
+   var awsCredentials = new MongoCredential("MONGODB-AWS", new MongoExternalAwsIdentity(), new ExternalEvidence());
+
+   var mongoClientSettings = MongoClientSettings.FromConnectionString("mongodb+srv://<atlasUri>");
+   mongoClientSettings.Credential = awsCredentials;
+   var client = new MongoClient(mongoClientSettings);
+
+To specify the authentication mechanism and authentication source in the connection 
+string, add them as parameters as shown in the following example:
+
+.. code-block:: csharp
+
+   var connectionString = "mongodb+srv://<atlasUri>?authMechanism=MONGODB-AWS&authSource=$external";
+   var mongoClientSettings = MongoClientSettings.FromConnectionString(connectionString);
+   var client = new MongoClient(mongoClientSettings);
+
+After specifying the authentication mechanism and source, you must set the 
+environment variables required for the type of authentication the driver will perform. 
+The next examples show how to provide your credentials by 
+setting environment variables for the following types of authentication:
+
+* Programmatic access keys
+* ECS container credentials
+* EC2 container credentials
+* EKS container credentials
+
+.. important::
+
+   Before checking environment variables for AWS IAM credentials, the {+driver-short+} 
+   will look for credentials in the following places:
+
+   * App.config and web.config files for the project
+   * Web identity provider
+   * Local AWS configuration
+
+The following examples show how you can set your **programmatic access keys**
+in environment variables by using ``bash`` or a similar shell:
+
+.. code-block:: bash
+
+   export AWS_ACCESS_KEY_ID=<awsKeyId>
+   export AWS_SECRET_ACCESS_KEY=<awsSecretKey>
+   export AWS_SESSION_TOKEN=<awsSessionToken>
+
+.. note::
+   
+   Omit the line containing ``AWS_SESSION_TOKEN`` if you don't need an AWS
+   session token for that role.
+
+To authenticate by using **ECS container credentials**, set the ECS-endpoint-relative 
+URI in an environment variable by using ``bash``
+or a similar shell as shown in the following example:
+
+.. code-block:: bash
+
+   export AWS_CONTAINER_CREDENTIALS_RELATIVE_URI=<your ECS endpoint>
+
+To authenticate by using **EC2 container credentials**, make sure none of the
+aforementioned environment variables are set. The driver obtains the
+credentials from the default IPv4 EC2 instance metadata endpoint.   
+
+To authenticate by using **EKS cluster credentials**,
```

Comment:
Dima said it would be imprecise xo call these "EKS credentials" since they can be used with other services too. I've changed the name on this page but will mention it can be used with EKS.